### PR TITLE
Vendor kubernetes-1.16.0

### PR DESCRIPTION
**Pull request by ialidzhikov**
**What this PR does / why we need it**:
Currently trying to vendor kubernetes-1.16.0 in gardener/gardener, one of the build errors is
```
# github.com/gardener/controller-manager-library/pkg/resources
vendor/github.com/gardener/controller-manager-library/pkg/resources/clients.go:45:39: undefined: serializer.DirectCodecFactory
```

kubernetes/kubernetes#76098 introduces a backwards incompatible change renaming `DirectCodecFactory` to `WithoutConversionCodecFactory`.

**Which issue(s) this PR fixes**:
Prerequisite for gardener/gardener#1206

**Special notes for your reviewer**:

**Release note**:

```improvement operator
Kubernetes dependencies are updated to `kubernetes-1.16.0`.
```

### CODE REVIEW

File: pkg/controllermanager/controller/configure.go [hunk 1]
```diff
@@ -260,7 +260,7 @@ func (this Configuration) Name(name string) Configuration {
 	return this
 }
 
-func (this Configuration) Require(names ... string) Configuration {
+func (this Configuration) Require(names ...string) Configuration {
 names:
```
Comment (vpnachev) [inline pkg/controllermanager/controller/configure.go]: btw, what is this ...

Comment (ialidzhikov) [reply pkg/controllermanager/controller/configure.go]: Ref https://gobyexample.com/variadic-functions

Comment (vpnachev) [reply pkg/controllermanager/controller/configure.go]: I meaned the `names:` thing on line 264

Comment (ialidzhikov) [reply pkg/controllermanager/controller/configure.go]: Ref https://www.ardanlabs.com/blog/2013/11/label-breaks-in-go.html

**Comment by MartinWeindel** [comment]
@ialidzhikov @vpnachev Thanks!

**Pull request merged**